microprofile/fault-tolerance/src/main/java/io/helidon/microprofile/faulttolerance/MethodInvoker.java

```diff
@@ -315,15 +298,8 @@ public boolean cancel(boolean mayInterruptIfRunning) {
         handler = createMethodHandler(methodState);
 
         // Gather information about current request scope if active
-        try {
-            requestController = CDI.current().select(RequestContextController.class).get();
-            requestScope = CDI.current().select(RequestScope.class).get();
-            requestContext = requestScope.referenceCurrent();
-        } catch (Exception e) {
-            requestScope = null;
-            LOGGER.fine(() -> "Request context not active for method " + method
-                    + " on thread " + Thread.currentThread().getName());
-        }
+        requestScopeHelper = new RequestScopeHelper();
+        requestScopeHelper.saveScope();
```

Comment:
Would we ever legitimately instantiate a helper without also saving the scope right away? 

If not, maybe fold the save work into the constructor? Or make the constructor private and use a static factory method that instantiates the helper and then invokes its `saveScope` method?

That would not work if there are ever cases where a single helper instance is created and then reused through several `save` and `clear` cycles.

Reply:
I don't have a use case for saving/clearing/saving yet, so it is a bit more generic than necessary. Having said that, I think it easier to read with an explicit call to `saveScope`.

Reply:
My concern was that correct usage requires me to remember to both instantiate the helper and invoke `saveScope` explicitly. This is internal so I'm not overly worried.